lib/runners/processor/metrics_codeclone.rb

```diff
@@ -0,0 +1,83 @@
+module Runners
+  class Processor::MetricsCodeClone < Processor
+
+    Schema = _ = StrongJSON.new do
+      # @type self: SchemaClass
+
+      let :issue, object(
+        clones: integer,
+      )
+    end
+
+    def initialize(**params)
+      super(**params)
+      @pmd_cpd = PmdCpd.new(**params)
+    end
+
+    def config_linter
+      @pmd_cpd.config_linter
+    end
+
+    def analyzer_bin
+      @pmd_cpd.analyzer_bin
+    end
+
+    def analyzer_version
+      @pmd_cpd.analyzer_version
+    end
+
+    def check_root_dir_exist
+      @pmd_cpd.check_root_dir_exist
+    end
+
+    def show_runtime_versions
+      @pmd_cpd.show_runtime_versions
+    end
+
+    def setup(&block)
+      @pmd_cpd.setup(&block)
+    end
+
+    def use_git_metadata_dir?
+      @pmd_cpd.use_git_metadata_dir?
+    end
+
+    def analyze(changes)
+      result = @pmd_cpd.analyze(changes)
+      @pmd_cpd.warnings.each { |warn| @warnings << warn }
+      return result unless result.is_a? Results::Success
+
+      issues = result.issues
+      file_issues =
+        issues
+          .map { |issue| issue.path }
+          .uniq
+          .map { |filepath| construct_file_issue(issues, filepath) }
+
+      Results::Success.new(guid: guid, analyzer: analyzer, issues: file_issues)
+    end
+
+    private
+
+    def construct_file_issue(issues, filepath)
+      issues_in_file = issues.select { |issue| issue.path == filepath }
+      clones = issues_in_file.length
+      sum_of_lines = issues_in_file.inject(0) do |sum, issue|
+        issue_obj = issue.object or raise "Required object: #{issue.inspect}"
+        sum + issue_obj[:lines]
+      end
```

Comment:
[suggestion] `Array#sum` with a block may be able to make the code clear.

https://ruby-doc.org/core-2.7.2/Array.html#sum-method

```suggestion
      sum_of_lines = issues_in_file.sum do |issue|
        issue_obj = issue.object or raise "Required object: #{issue.inspect}"
        issue_obj[:lines]
      end
```

Reply:
Fixed. commit e6f649e8dc57a69445f5074e76d88b8a0333c611